app/helpers/sessions_helper.rb

```diff
@@ -1,17 +1,41 @@
 module SessionsHelper
   def log_in(user)
     session[:user_id] = user.id
+    session[:cart] = {}
+    @cart = session[:cart]
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.encrypted[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
   end
 
   def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.encrypted[:user_id])
+      user = User.find_by(id: user_id)
+      if user&.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
     @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
   end
 
   def logged_in?
     !current_user.nil?
```

Comment:
I think it will be more readable if you use `.present?` instead of negating the `nil?`